edge-apps/helpers/tfl/bus-stop-lookup/script.js

```diff
@@ -1,85 +1,566 @@
+              .map(row => {
+                return {
+                  'Stop Name': stripSpecialCharacters(row[stopNameField] || '').trim(),
+                  'Stop Code': row[stopCodeField] || ''
+                }
+              })
+
+            // Sort the data alphabetically by stop name
+            fullData.sort((a, b) => a['Stop Name'].localeCompare(b['Stop Name']))
+
+            // Initialize filtered data with full data
+            filteredData = [...fullData]
+
+            // Hide loading indicator
+            if (loadingIndicator) {
+              loadingIndicator.style.display = 'none'
             }
-          })
-
-          console.log('Filtered Data:', filteredData) // Log filtered data
-          renderTable(filteredData)
-          addSearchFunctionality(filteredData)
-        },
-        error: function (error) {
-          console.error('Error parsing CSV:', error)
-        }
-      })
+
+            // Create filters
+            createFilters()
+
+            renderPagination()
+            renderTablePage(currentPage)
+            addSearchFunctionality()
+          },
+          error: function (error) {
+            console.error('Error parsing CSV:', error)
+            showErrorState('Error processing data. Please try again later.')
+          }
+        })
+      }, 100) // Small delay to allow UI to update
     })
     .catch(error => {
       console.error('Error fetching CSV:', error)
+      showErrorState('Could not load bus stop data. Please check your connection and try again.')
+    })
+
+  function setupPageStructure () {
+    // Create loading indicator if it doesn't exist
+    if (!document.getElementById('loading')) {
+      const loadingDiv = document.createElement('div')
+      loadingDiv.id = 'loading'
+      loadingDiv.className = 'text-center p-4'
+      loadingDiv.innerHTML = '<p class="text-xl">Loading bus stop data...</p><div class="loader mt-2"></div>'
+
+      // Find the data-container or create it if it doesn't exist
+      let dataContainer = document.querySelector('.data-container')
+      if (!dataContainer) {
+        dataContainer = document.createElement('div')
+        dataContainer.className = 'data-container min-h-[400px]'
+
+        // Make sure the table wrapper exists
+        const tableWrapper = document.querySelector('.relative.overflow-x-auto') ||
+          document.querySelector('.overflow-x-auto')
+
+        if (tableWrapper) {
+          // Check if the table already exists
+          if (!tableWrapper.querySelector('table')) {
+            const table = document.createElement('table')
+            table.className = 'min-w-full bg-white'
+            table.innerHTML = '<thead><tr></tr></thead><tbody></tbody>'
+            dataContainer.appendChild(table)
+            tableWrapper.appendChild(dataContainer)
+          } else {
+            tableWrapper.appendChild(dataContainer)
+          }
```

Comment:
```suggestion
            tableWrapper.appendChild(dataContainer)
          }
          tableWrapper.appendChild(dataContainer)
```

Reply:
`tableWrapper.appendChild(dataContainer)` is both in if and else. The statement can be moved outside and else block can be removed